### PR TITLE
Update Kubeflow Notebooks docs

### PR DESCRIPTION
**Pull request by thesuperzapper**
__This PR updates the Kubeflow Notebooks docs:__
- creates new "overview" page
- removes specific references to "Jupyter" (as Kubeflow 1.3 now supports multiple IDEs)
- added lots of information about how to build your own container images (for Jupyter, VSCode, and RStudio)
- simplifies the "getting started" guide (and makes it generic to not just be about Jupyter)

__This is the first step towards having the docs updated for 1.4:__
- https://github.com/kubeflow/manifests/issues/1961

__In future PRs, to be updated for 1.3/1.4, we need to:__
- the PNG images of the UI need to be updated for 1.4
- add a page about `spawner_ui_config.yaml`
- add a page about how to configure logos for "group 1" and "group 2"
    - `/src/apps/default/static/assets/group-one-icon.svg`
    - `/src/apps/default/static/assets/group-one-logo.svg`
    - `/src/apps/default/static/assets/group-two-icon.svg`
    - `/src/apps/default/static/assets/jupyter-icon.svg`
    - `/src/apps/default/static/assets/jupyterlab-logo.svg`

### CODE REVIEW

**Comment by thesuperzapper** [comment]
@RFMVasconcelos or @kubeflow/wg-notebooks-leads can you please review?

**Comment by thesuperzapper** [comment]
@zijianjoy can you please review this before we cut the 1.4 branch (it's pretty important updates for the Kubeflow Notebooks docs)!

**Comment by kimwnasptd** [comment]
@thesuperzapper yep, it's one of my highest priorities! I'm having my hands full with KubeCon right now, but I'll try to do this in parallel in the next days

File: content/en/docs/components/notebooks/jupyter-tensorflow-examples.md [hunk 1]
```diff
@@ -0,0 +1,59 @@
++++
+title = "Jupyter TensorFlow Examples"
+description = "Examples using Jupyter and TensorFlow in Kubeflow Notebooks"
+weight = 40
+
++++
+
+## Mnist Example
```
Comment (shannonbradshaw) [inline content/en/docs/components/notebooks/jupyter-tensorflow-examples.md:8]: It would be helpful to have a brief intro paragraph explaining what this paragraph is doing.

File: content/en/docs/components/notebooks/quickstart-guide.md [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+    - __Standard image__: Click the __"Image"__ dropdown menu to see the list of available images.
+      (You can choose from the list configured by your Kubeflow administrator)
+
+7. Specify the amount of __"CPU"__ that your notebook server will request.
+
+8. Specify the amount of __"RAM"__ that your notebook server will request.
+
+9. Specify a __"workspace volume"__ to be mounted as a PVC Volume on your home folder.
+
+10. *(Optional)* Specify one or more __"data volumes"__ to be mounted as a PVC Volumes.
```
Comment (shannonbradshaw) [inline content/en/docs/components/notebooks/quickstart-guide.md:62]: I'd recommend including a screenshot for workspace and data volumes.

File: content/en/docs/components/notebooks/quickstart-guide.md [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+
+11. *(Optional)* Specify one or more additional __"configurations"__
```
Comment (shannonbradshaw) [inline content/en/docs/components/notebooks/quickstart-guide.md:64]: And one for configurations. Given the number of options, it's easy for people to get lost. 

I realize we want to focus users on the required steps, but I think these screenshots add clarity.

File: content/en/docs/components/notebooks/quickstart-guide.md [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+    - Currently, there is no implementation in Kubernetes to activate shared memory.
+    - As a workaround, Kubeflow mounts an empty directory volume at `/dev/shm`.
+
+14. Click __"LAUNCH"__ to create a new Notebook CRD with your specified settings.
+    - You should see an entry for your new notebook server on the __"Notebook Servers"__ page
+    - There should be a spinning indicator in the __"Status"__ column.
+    - It can take a few minutes for kubernetes to provision the notebook server pod.
+    - You can check the status of your Pod by hovering your mouse cursor over the icon in the __"Status"__ column.
+
+15. Click __"CONNECT"__ to view the web interface exposed by your notebook server.
```
Comment (shannonbradshaw) [inline content/en/docs/components/notebooks/quickstart-guide.md:103]: Nit. Maybe include a red arrow for the Connect link as we did above.

File: content/en/docs/components/notebooks/quickstart-guide.md [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+12. *(Optional)* Specify any __"GPUs"__ that your notebook server will request.
+    - Kubeflow uses "limits" in Pod requests to provision GPUs onto the notebook Pods
+      (Details about scheduling GPUs can be found in the [Kubernetes Documentation](https://kubernetes.io/docs/tasks/manage-gpus/scheduling-gpus/).)
+
+13. *(Optional)* Specify the setting for __"enable shared memory"__.
+    - Some libraries like PyTorch use shared memory for multiprocessing.
+    - Currently, there is no implementation in Kubernetes to activate shared memory.
+    - As a workaround, Kubeflow mounts an empty directory volume at `/dev/shm`.
+
+14. Click __"LAUNCH"__ to create a new Notebook CRD with your specified settings.
```
Comment (shannonbradshaw) [inline content/en/docs/components/notebooks/quickstart-guide.md:97]: Since this button is hidden at the bottom until you scroll, I'd suggest a screenshot for this as well.

**Comment by shannonbradshaw** [comment]
Left a few comments, but the overall structure and additions look good to me. Would you like to get a review from @kimwnasptd as well.

Thanks @thesuperzapper ! Really appreciate the work here.

File: content/en/docs/components/notebooks/jupyter-tensorflow-examples.md [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+
+    ```
+    Accuracy:  0.9012
+    ```
+
+## Next steps
+
+- See a [simple example](https://github.com/kubeflow/examples/tree/master/pipelines/simple-notebook-pipeline) of creating Kubeflow pipelines in a Jupyter notebook.
+- Build machine-learning pipelines with the [Kubeflow Pipelines SDK](/docs/components/pipelines/sdk/sdk-overview/).
+- Explore [Kubeflow Fairing](/docs/external-add-ons/fairing/) for a complete solution to building, training, and deploying an ML model from a notebook.
```
Comment (kimwnasptd) [inline content/en/docs/components/notebooks/jupyter-tensorflow-examples.md:58]: Let's not recommend Fairing here.
It has not been maintained for more than a year and its docs are out of date. It also does not currently belong to any WG

File: content/en/docs/components/notebooks/jupyter-tensorflow-examples.md [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+- Learn the advanced features available from a Kubeflow notebook, such as [submitting Kubernetes resources](/docs/components/notebooks/submit-kubernetes/) or [building Docker images](/docs/components/notebooks/custom-notebook/).
```
Comment (kimwnasptd) [inline content/en/docs/components/notebooks/jupyter-tensorflow-examples.md:59]: Let's rephrase this one a little bit as well:

```
Learn the advanced features available in the example notebook images, such as ...
```

**Review by kimwnasptd — CHANGES_REQUESTED** [review]
Thank you very much for this work @thesuperzapper! I've left some initial comments on some parts of your work. The main theme of the comments is to not use the naming `Kubeflow Notebooks` since this implies that only some Notebooks are Kubeflow while others are not.

I'll also do a review of the `overview`, `quickstart-guide`, `submit-kubernetes`, `troubleshooting` pages tomorrow as well.

**Comment by thesuperzapper** [comment]
> Thank you very much for this work @thesuperzapper! I've left some initial comments on some parts of your work. The main theme of the comments is to not use the naming `Kubeflow Notebooks` since this implies that only some Notebooks are Kubeflow while others are not.

@kimwnasptd I think I am not understanding what you mean by "this implies that only some Notebooks are Kubeflow while others are not".

In my mind, the name "Kubeflow Notebooks" is the __brand-name__ of the "applications which run notebooks in Kubeflow". Similarly, "Kubeflow Pipelines" is the __brand-name__ of the "set of applications that run pipelines in Kubeflow" (note, we already use "Kubeflow Pipelines" in the docs for that component).

To put it another way, these docs are describing the "set of applications in Kubeflow which run notebooks", and the collective noun for those things is "Kubeflow Notebooks" not "Notebooks".

I feel very strongly that this will improve the brand recognition of the Kubeflow Notebooks "product".

**Comment by StefanoFioravanzo** [comment]
@thesuperzapper @kimwnasptd My two cents here on naming. So the problem stems from the fact that the component’s name is the same name as the objects being handled by the component. I think this is why it’s `Kubeflow Pipelines` and not just `Pipelines`. Because `Pipelines` is a generic term that by itself does not identify a specific product. Take `Katib` or `KFServing` instead, they are brand names that identify a specific component.

So, in terms of Docs, I think that using `Kubeflow Notebooks` to refer to the components and `Notebooks` to the specific instances can help. **BUT**, this has in no way change what we do in the dashboard. There, we should continue showing `Notebooks`, which are the **objects**.

**Comment by thesuperzapper** [comment]
> @thesuperzapper @kimwnasptd My two cents here on naming. So the problem stems from the fact that the component’s name is the same name as the objects being handled by the component. I think this is why it’s `Kubeflow Pipelines` and not just `Pipelines`. Because `Pipelines` is a generic term that by itself does not identify a specific product. Take `Katib` or `KFServing` instead, they are brand names that identify a specific component.

@StefanoFioravanzo If I am reading this correctly, you are agreeing with my proposal to use the "brand name" of `"Kubeflow Notebooks"` on https://kubeflow.org, rather than what is currently on the website: `"Notebook Servers"`?

> So, in terms of Docs, I think that using `Kubeflow Notebooks` to refer to the components and `Notebooks` to the specific instances can help. **BUT**, this has in no way change what we do in the dashboard. There, we should continue showing `Notebooks`, which are the **objects**.

I completely agree that within Central Dashboard, it makes no sense to specify `"Kubeflow Notebooks"` on the sidebar, and we should continue to use `"Notebooks"`. However, this PR is changing the section-name on https://kubeflow.org from `"Notebook Servers"` to `"Kubeflow Notebooks"` (which will not affect Central Dashboard).

File: content/en/docs/components/notebooks/jupyter-tensorflow-examples.md (hunk 6 above)
Comment (shannonbradshaw) [reply content/en/docs/components/notebooks/jupyter-tensorflow-examples.md:58]: +1

File: content/en/docs/components/notebooks/container-images.md [hunk 8]
```diff
@@ -0,0 +1,73 @@
...
+[codeserver](https://github.com/kubeflow/kubeflow/tree/master/components/example-notebook-servers/codeserver) | [`public.ecr.aws/j1r0q0g6/notebooks/notebook-servers/codeserver:{TAG}`](https://gallery.ecr.aws/j1r0q0g6/notebooks/notebook-servers/codeserver) | base [code-server](https://github.com/cdr/code-server) (Visual Studio Code) image
+[jupyter](https://github.com/kubeflow/kubeflow/tree/master/components/example-notebook-servers/jupyter) | [`public.ecr.aws/j1r0q0g6/notebooks/notebook-servers/jupyter:{TAG}`](https://gallery.ecr.aws/j1r0q0g6/notebooks/notebook-servers/jupyter) | base [JupyterLab](https://github.com/jupyterlab/jupyterlab) image
+[rstudio](https://github.com/kubeflow/kubeflow/tree/master/components/example-notebook-servers/rstudio) | [`public.ecr.aws/j1r0q0g6/notebooks/notebook-servers/rstudio:{TAG}`](https://gallery.ecr.aws/j1r0q0g6/notebooks/notebook-servers/rstudio) | base [RStudio](https://github.com/rstudio/rstudio) image
+
+### Full Images
+
+These images extend the [base images](#base-images) with common packages used in the real world.
```
[the remark this answers is not in the document]
Comment (shannonbradshaw) [reply content/en/docs/components/notebooks/container-images.md:28]: +1

**Comment by shannonbradshaw** [comment]
@kimwnasptd you gave a thumbs up to @StefanoFioravanzo comment above. Are you in agreement with @thesuperzapper's summary that follows that comment?

File: content/en/docs/components/notebooks/_index.md [hunk 9]
```diff
@@ -1,5 +1,5 @@
 +++
-title = "Notebook Servers"
-description = "Using Jupyter notebooks in Kubeflow"
+title = "Kubeflow Notebooks"
```
[the remark this answers is not in the document]
Comment (thesuperzapper) [reply content/en/docs/components/notebooks/_index.md:2]: As discussed here, https://github.com/kubeflow/website/pull/3003#issuecomment-960241069, we will use "Kubeflow Notebooks"

File: content/en/docs/components/notebooks/container-images.md [hunk 10]
```diff
@@ -0,0 +1,73 @@
++++
+title = "Container Images"
+description = "About Container Images for Kubeflow Notebooks"
```
[the remark this answers is not in the document]
Comment (thesuperzapper) [reply content/en/docs/components/notebooks/container-images.md:3]: As discussed here, https://github.com/kubeflow/website/pull/3003#issuecomment-960241069, we will use "Kubeflow Notebooks"

File: content/en/docs/components/notebooks/_index.md [hunk 11]
```diff
@@ -1,5 +1,5 @@
...
+description = "Documentation for Kubeflow Notebooks"
```
[the remark this answers is not in the document]
Comment (thesuperzapper) [reply content/en/docs/components/notebooks/_index.md:3]: As discussed here, https://github.com/kubeflow/website/pull/3003#issuecomment-960241069, we will use "Kubeflow Notebooks"

File: content/en/docs/components/notebooks/container-images.md [hunk 12]
```diff
@@ -0,0 +1,73 @@
...
+weight = 30
+                    
++++
+Kubeflow Notebooks natively supports three types of notebooks, [JupyterLab](https://github.com/jupyterlab/jupyterlab), [RStudio](https://github.com/jupyterlab/jupyterlab), and [Visual Studio Code (code-server)](https://github.com/cdr/code-server), but any web-based IDE should work.
```
[the remark this answers is not in the document]
Comment (thesuperzapper) [reply content/en/docs/components/notebooks/container-images.md:7]: As discussed here, https://github.com/kubeflow/website/pull/3003#issuecomment-960241069, we will use "Kubeflow Notebooks"

[the remark this answers is not in the document]
Comment (thesuperzapper) [reply content/en/docs/components/notebooks/container-images.md:7]: In this case, we are using the term RStudio to refer to the software itself, which should be fine.

File: content/en/docs/components/notebooks/container-images.md [hunk 13]
```diff
@@ -0,0 +1,73 @@
...
+Notebook servers run as containers inside a Kubernetes Pod, which means the type of IDE (and which packages are installed) is determined by the Docker image you pick for your server.
+
+## Images
+
+We provide a number of [example container images](https://github.com/kubeflow/kubeflow/tree/master/components/example-notebook-servers) to get you started.
+
+### Base Images
+
+These images provide a common starting point for Kubeflow Notebook containers.
```
[the remark this answers is not in the document]
Comment (thesuperzapper) [reply content/en/docs/components/notebooks/container-images.md:16]: As discussed here, https://github.com/kubeflow/website/pull/3003#issuecomment-960241069, we will use "Kubeflow Notebooks"

File: content/en/docs/components/notebooks/jupyter-tensorflow-examples.md [hunk 14]
```diff
@@ -0,0 +1,59 @@
++++
+title = "Jupyter TensorFlow Examples"
+description = "Examples using Jupyter and TensorFlow in Kubeflow Notebooks"
```
[the remark this answers is not in the document]
Comment (thesuperzapper) [reply content/en/docs/components/notebooks/jupyter-tensorflow-examples.md:3]: As discussed here, https://github.com/kubeflow/website/pull/3003#issuecomment-960241069, we will use "Kubeflow Notebooks"

File: content/en/docs/components/notebooks/jupyter-tensorflow-examples.md (hunk 7 above)
Comment (thesuperzapper) [reply content/en/docs/components/notebooks/jupyter-tensorflow-examples.md:59]: As discussed here, https://github.com/kubeflow/website/pull/3003#issuecomment-960241069, we will use "Kubeflow Notebooks"

File: content/en/docs/components/notebooks/jupyter-tensorflow-examples.md (hunk 6 above)
Comment (thesuperzapper) [reply content/en/docs/components/notebooks/jupyter-tensorflow-examples.md]: Resolved by https://github.com/kubeflow/website/pull/3003/commits/d91928b2d9633cbad49c8a9275a8bc40b5c488c5

File: content/en/docs/components/notebooks/container-images.md [hunk 15]
```diff
@@ -0,0 +1,73 @@
...
+
+<img src="/docs/images/notebook-container-image-chart.png" 
+     alt="A flow-chart showing how notebook container images depend on each other"  
+     class="mt-3 mb-3 border border-info rounded">
+
+## Custom Images
+
+Packages that are installed by users __after spawning__ a notebook server will __only last the lifetime of the Pod__ unless those packages are installed into a PVC-backed directory.
+
+We recommend extending the [base images](#base-images) with the packages and dependencies you need. (See [a guide](https://github.com/kubeflow/kubeflow/tree/master/components/example-notebook-servers#custom-images) that explains how to do this.)
```
[the remark this answers is not in the document]
Comment (thesuperzapper) [reply content/en/docs/components/notebooks/container-images.md]: Resolved by https://github.com/kubeflow/website/pull/3003/commits/d91928b2d9633cbad49c8a9275a8bc40b5c488c5

File: content/en/docs/components/notebooks/container-images.md (hunk 8 above)
[the remark this answers is not in the document]
Comment (thesuperzapper) [reply content/en/docs/components/notebooks/container-images.md]: Resolved by https://github.com/kubeflow/website/pull/3003/commits/d91928b2d9633cbad49c8a9275a8bc40b5c488c5

File: content/en/docs/components/notebooks/container-images.md (hunk 12 above)
[the remark this answers is not in the document]
Comment (thesuperzapper) [reply content/en/docs/components/notebooks/container-images.md]: Resolved by https://github.com/kubeflow/website/pull/3003/commits/d91928b2d9633cbad49c8a9275a8bc40b5c488c5

**Comment by thesuperzapper** [comment]
@shannonbradshaw thanks for your comments, but in the interest of getting this PR merged, can you raise a separate PR which makes your desired changes after this has been merged (NOTE: the `notebooks/quickstart-guide.md` is not actually a new page, it was just moved/cleaned by this PR)

**Comment by thesuperzapper** [comment]
@Bobgy this PR has been reviewed by @shannonbradshaw and @kimwnasptd and I have addressed the comments from them, but we still need someone with root OWNER to `/approve` this (because it updates the `_redirects` file)

Can you `/approve` this so that we can have the notebooks docs updated for Kubeflow 1.4 (and 1.3, lol)?

**Comment by shannonbradshaw** [comment]
@thesuperzapper it is OK with me to address my comments in a separate PR. Thanks for following up. It looks like we've provided enough time for other reviewers to chime back in. I'm in favor of merging this. @Bobgy or @zijianjoy

**Comment by Bobgy** [comment]
/approve
I'm interested to see if there are ways for WGs to self service on redirection changes.

Looks like hugo also supports a field called "aliases", it can be added directly on content pages.
Can you confirm if you can set up redirect in that approach?
https://gohugo.io/content-management/urls/#example-aliases

**Comment by thesuperzapper** [comment]
@shannonbradshaw can you `/lgtm` so this gets merged?

**Comment by kimwnasptd** [comment]
I'll go on and merge this PR, thanks for your effort in this @thesuperzapper!

Although, in the future please give an explicit heads up to WG leads when renaming the product name of a Component since this needs discussion. This is a significant change and there was no mention of this the PR description. Doing such a change alongside a whole refactoring makes it difficult to discuss specific parts when dealing with such huge PRs.

/lgtm
